throttler.go

```diff
@@ -62,26 +64,26 @@ func NewBatchedThrottler(maxWorkers, batchingTotal, batchSize int) *Throttler {
 // all of the jobs have been dispatched. It stops blocking when Done has been called
 // as many times as totalJobs.
 func (t *Throttler) Throttle() int {
-	if t.totalJobs < 1 {
-		return t.errorCount
+	if atomic.LoadInt32(&t.totalJobs) < 1 {
+		return int(atomic.LoadInt32(&t.errorCount))
 	}
-	t.jobsStarted++
-	t.workerCount++
+	atomic.AddInt32(&t.jobsStarted, 1)
+	atomic.AddInt32(&t.workerCount, 1)
 
-	if t.workerCount == t.maxWorkers {
+	if atomic.LoadInt32(&t.workerCount) == atomic.LoadInt32(&t.maxWorkers) {
+		atomic.AddInt32(&t.jobsCompleted, 1)
+		atomic.AddInt32(&t.workerCount, -1)
```

Comment:
Was this switch of order on purpose?

Reply:
Yes, it was causing a race condition here and where it was fixed in #7.